src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -2345,7 +2345,12 @@ private static DType SupertypeAggregateCore(DType type1, DType type2, bool useLe
             }
 
             Contracts.Assert(!fError);
-            return new DType(type1.Kind, treeRes, UnionDataSourceInfoMetadata(type1, type2));
+            var returnType = new DType(type1.Kind, treeRes, UnionDataSourceInfoMetadata(type1, type2), type1.DisplayNameProvider);
```

Comment:
Wondering ... are there other similar bugs here for other type algebra operations? 
 Ie, any place that we call UnionDataSourceInfoMetadata without also checking DisplayNameProvider?